expression/builtin_math.go

```diff
@@ -140,6 +141,45 @@ func builtinCeil(args []types.Datum, ctx context.Context) (d types.Datum, err er
 	return
 }
 
+type floorFunctionClass struct {
+	baseFunctionClass
+}
+
+func (c *floorFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
+	return &builtinFloorSig{newBaseBuiltinFunc(args, ctx)}, errors.Trace(c.verifyArgs(args))
+}
+
+type builtinFloorSig struct {
+	baseBuiltinFunc
+}
+
+func (b *builtinFloorSig) eval(row []types.Datum) (types.Datum, error) {
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return types.Datum{}, errors.Trace(err)
+	}
+	return builtinFloor(args, b.ctx)
+}
+
+// See http://dev.mysql.com/doc/refman/5.7/en/mathematical-functions.html#function_floor
+func builtinFloor(args []types.Datum, ctx context.Context) (d types.Datum, err error) {
+	if args[0].IsNull() ||
+		args[0].Kind() == types.KindUint64 || args[0].Kind() == types.KindInt64 {
+		return args[0], nil
+	}
+
+	sc := ctx.GetSessionVars().StmtCtx
+	sc.IgnoreTruncate = true
+	f, err := args[0].ToFloat64(sc)
+	if err != nil {
+		sc.IgnoreTruncate = false
+		return d, errors.Trace(err)
+	}
+	sc.IgnoreTruncate = false
```

Comment:
`FLOOR(-1.b2) = -1` This will cause ErrTruncated error. Hence, sc.IgnoreTruncate has false value. I need to set sc.IgnoreTruncate to true in order to avoid ErrTruncated error which is came from getValidFloatPrefix(sc, str) in https://github.com/pingcap/tidb/blob/master/util/types/convert.go#L242